src/main/java/net/neoforged/neoforge/server/command/TPSCommand.java

```diff
@@ -60,4 +60,65 @@ private static long mean(long[] values) {
             sum += v;
         return sum / values.length;
     }
+
+    private static Component createComponent(MinecraftServer server, @Nullable ServerLevel dimension) {
+        MutableComponent prefix;
+        long[] times;
+        TickRateManager tickRateManager;
+
+        if (dimension == null) {
+            prefix = Component.translatable("commands.neoforge.tps.summary.overall");
+            times = server.getTickTimesNanos();
+            tickRateManager = server.tickRateManager();
+        } else {
+            var dimKey = dimension.dimension();
+            var dimensionTimes = server.getTickTime(dimKey);
+
+            prefix = Component.translatable("commands.neoforge.tps.summary.dimension", createDimensionComponent(dimension));
+            times = dimensionTimes == null ? UNLOADED : dimensionTimes;
+            tickRateManager = dimension.tickRateManager();
```

Comment:
I don't think this matters, since that method just returns the server's tick rate manager. I leave it up to you if you want to remove this line and just use the tick rate manager obtained directly from the server instance.

Reply:
You are correct. It does retrieve the servers tick rate manager, but what if some mod allowed the tick rate manager to be per dimension rather than server wide, this would already support that.
It also just feels more correct imo to pull off of the dimension instance if one exists rather than the server.